features/step_definitions/membership_application_steps.rb

```diff
@@ -25,18 +27,28 @@
     contact_email = hash['contact_email'] && ! hash['contact_email'].empty? ?
                     hash['contact_email'] : hash[:user_email]
 
+    legacy_app = hash['legacy'] == 'true' ? true : false
+
     if (ma = user.shf_application)
 
       user.shf_application.companies << companies
 
     else
       num_categories = hash[:categories] ? 0 : 1
 
-      ma = FactoryBot.build(:shf_application,
-                            attributes.merge(user: user,
-                            contact_email: contact_email,
-                            create_company: false,
-                            num_categories: num_categories))
+      if legacy_app
```

Comment:
I would push this responsibility into the FactoryBot:  Let the FactoryBot create the appropriate objects based on data you pass into it, instead of having the responsibility at this level.  (That's what the factory bot is supposed to be doing, even though we don't always use that appropriately.)
Instead of using a `trait` I would pass in a transient variable and then have the FactoryBot handle it in the `after_build` code.
ex:  ` ma = FactoryBot.build(:shf_application, shf_attributes, skip_file_delivery_validation)`

And in the SHF application factory:

```
...
 transient do
    ...
    skip_file_delivery_validation { false }
  end

...

  after(:build) do |shf_app, evaluator|
    ...
    if evaluator.skip_file_delivery_validation
        < create/ build the right kind of thing or do what is needed >
    end

  end

Reply:
I need to interrupt the standard create action. Not sure how this could be done with transient.

Reply:
Ya - I see that now.  Having to do `instance.save(validate: false)` makes it tricky.  Thanks for pointing that out.

To me, the fact that you've having to put in all this code (all the conditionals) to skip validation is a _smell_ that points to some re-design needed to handle this.  (Seems like this is essentially a Null Object pattern.) BUT -- I would also say that doing that is definitely **not** something that should be done now.  (And perhaps when reviewed with all of the other priorities, should never be done -- just not important enough.) But seeing it makes my OO brain. ;-)